app/models/katello/system.rb

```diff
@@ -75,6 +75,12 @@ class System < Katello::Model
     scoped_search :on => :name, :complete_value => true
     scoped_search :in => :environment, :on => :organization_id, :complete_value => true, :rename => :organization_id
 
+    has_many :fact_values, :through => :foreman_host
+    has_many :fact_names, :through => :fact_values
+
+    scoped_search :in => :fact_values, :on => :value, :in_key=> :fact_names, :on_key=> :name, :rename => :facts, :complete_value => true,
```

Comment:
This will allow users to search in the UI/API using these facts but not autocomplete them in the UI -- do we want to expose that at this point?